geoutils/raster/delayed.py

```diff
@@ -0,0 +1,777 @@
+            "ystart": (starts[1][indexes_yi[i]] - map_depth[method]) * resolution[1],
+            "xres": resolution[0],
+            "yres": resolution[1],
+        }
+        for i in range(len(blocks))
+    ]
+
+    # Compute values delayed
+    list_interp = [
+        _delayed_interp_points_block(blocks[i], block_ids[i], points_arr[:, ind_per_block[i]])
+        for i, data_chunk in enumerate(blocks)
+        if len(ind_per_block[i]) > 0
+    ]
+
+    # We define the expected output shape and dtype to simplify things for Dask
+    list_interp_delayed = [
+        da.from_delayed(p, shape=(1, len(ind_per_block[i])), dtype=darr.dtype) for i, p in enumerate(list_interp)
+    ]
+    interp_points = np.concatenate(dask.compute(*list_interp_delayed), axis=0)
+
+    # Re-order per-block output points to match their original indices
+    indices = np.concatenate(ind_per_block).astype(int)
+    argsort = np.argsort(indices)
+    interp_points = np.array(interp_points)[argsort]
+
+    return interp_points
+
+
+# 3/ REPROJECT
+# Part of the code (defining a GeoGrid and GeoTiling classes) is inspired by
+# https://github.com/opendatacube/odc-geo/pull/88, modified to be concise, stand-alone and rely only on
+# Rasterio/GeoPandas
+# Could be submitted as a PR to Rioxarray? (but not sure the dependency to GeoPandas would work there)
+
+# We define a GeoGrid and GeoTiling class (which composes GeoGrid) to consistently deal with georeferenced footprints
+# of chunked grids
+GeoGridType = TypeVar("GeoGridType", bound="GeoGrid")
+
+
+class GeoGrid:
+    """
+    Georeferenced grid class.
+
+    Describes a georeferenced grid through a geotransform (one-sided bounds and resolution), shape and CRS.
+    """
+
+    def __init__(self, transform: rio.transform.Affine, shape: tuple[int, int], crs: rio.crs.CRS | None):
+
+        self._transform = transform
+        self._shape = shape
+        self._crs = crs
+
+    @property
+    def transform(self) -> rio.transform.Affine:
+        return self._transform
+
+    @property
+    def crs(self) -> rio.crs.CRS:
+        return self._crs
+
+    @property
+    def shape(self) -> tuple[int, int]:
+        return self._shape
+
+    @property
+    def height(self) -> int:
+        return self.shape[0]
+
+    @property
+    def width(self) -> int:
+        return self.shape[1]
+
+    @property
+    def res(self) -> tuple[int, int]:
+        return self.transform[0], abs(self.transform[4])
+
+    @property
+    def bounds(self) -> rio.coords.BoundingBox:
+        return rio.coords.BoundingBox(*rio.transform.array_bounds(self.height, self.width, self.transform))
+
+    def bounds_projected(self, crs: rio.crs.CRS = None) -> rio.coords.BoundingBox:
+        if crs is None:
+            crs = self.crs
+        return _get_bounds_projected(bounds=self.bounds, in_crs=self.crs, out_crs=crs)
+
+    @property
+    def footprint(self) -> gpd.GeoDataFrame:
```

Comment:
For easier maintenance, this function could simply call `footprint_projected`. Same for `bounds` and `bounds_projected`.

Reply:
Done